test/e2e/soak_namespace.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	//"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func extinguish(c *client.Client, totalNS int, maxAllowedAfterDel int, maxSeconds int) {
+
+	var err error
+
+	for n := 0; n < totalNS; n += 1 {
+		_, err = createTestingNS(fmt.Sprintf("nslifetest-%v", n), c)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	//deletion completed in maxSeconds
+	deletionCompletedTimeout := time.After(time.Duration(maxSeconds) * time.Second)
+
+	//start deleting after 10 seconds.
+	deletionStart := time.After(time.Duration(10 * time.Second))
+	tick := time.Tick(2 * time.Second)
+T:
+	for {
```

Comment:
hmm i guess the `wait.Poll()` is a good wy to do this instead of channels.  ill resubmit in a few .  thanks for this idea .